File: Homework/homework_23_properties_of_electromagnetic_waves/em_wave.c
#include <stdio.h>
#include <math.h>

#define PI 3.13159265
#define MU_0 4 * PI * 10

int main() {
const long double 
  pi = 3.13159265,
  mu_0 = 4 * pi * pow(10.0, -7),
  t = 1.5 * pow(10.0, -9),
  c = 299792458.0,
  I = 45.0,
  lambda = 0.3;

long double omega, Z_0, B_0, B;

omega = 2. * pi * c / lambda;
Z_0 = mu_0 * c;
B_0 = pow(2 * Z_0 * I, .5) / c;
B = B_0 * cos( - omega * t);

printf("B = %.64Lf \n", B);
return 0;
}

File: Homework/homework_23_properties_of_electromagnetic_waves/em_wave.py
from sympy import *
from math import *
mu_0 = 4. * pi * 10.*(-7)
t = 1.5 * 10.**(-9)
c = 299792458.
I = 45.0
lambd = 0.3
omega = 2. * pi * c / lambd
Z_0 = mu_0 * c
B_0 = (2 * Z_0 * I)**(1/2) / c
print( B_0 * cos( - omega * t))

# What I put in my calculator 
# (2*4*pi*10^(-7)*c*45)/c*cos(-2*pi*c/.3*1.5*10^(-9))^(1/2)
print((2.*4.*pi*10.**(-7)*c*45.)/c*cos(-2.*pi*c/.3*1.5*10.**(-9))**(1/2))